graylog2-server/src/main/java/org/graylog2/rest/resources/streams/StreamResource.java

```diff
@@ -189,6 +209,41 @@ private boolean ensureUserHasPermissionsForStream(User user, String id) throws V
         return permissionsChanged;
     }
 
+    @GET
+    @Timed
+    @Path("/page")
+    @ApiOperation(value = "Get a paginated list of streams")
+    @Produces(MediaType.APPLICATION_JSON)
+    public StreamPageListResponse getPage(@ApiParam(name = "page") @QueryParam("page") @DefaultValue("1") int page,
+        @ApiParam(name = "per_page") @QueryParam("per_page") @DefaultValue("50") int perPage,
+        @ApiParam(name = "query") @QueryParam("query") @DefaultValue("") String query,
+        @ApiParam(name = "sort",
+                value = "The field to sort the result on",
+                required = true,
+                allowableValues = "title,description")
+        @DefaultValue(StreamImpl.FIELD_TITLE) @QueryParam("sort") String sort,
+        @ApiParam(name = "order", value = "The sort direction", allowableValues = "asc, desc")
+        @DefaultValue("asc") @QueryParam("order") String order) {
+
+        SearchQuery searchQuery;
+        try {
+            searchQuery = searchQueryParser.parse(query);
+        } catch (IllegalArgumentException e) {
+            throw new BadRequestException("Invalid argument in search query: " + e.getMessage());
+        }
+        final Predicate<StreamDTO> permissionFilter = streamDTO -> isPermitted(RestPermissions.STREAMS_READ, streamDTO.id());
```

Comment:
We are duplicating a bit of code here vs. `StreamResource#get`. We have several options to address this:

  a) Reuse `getPage` in `get` and returning `getPage(...).streams()` in `get`. This reduces duplication to a minimum, but involves unnecessary boxing/instantiation.
  b) Extract a method that returns `streams` (of line 237), which is used in both methods.
  c) Extract `permissionFilter` and only reuse that one in both.